Mathlib/Topology/SubsetProperties.lean

```diff
@@ -2041,3 +2041,32 @@ theorem IsPreirreducible.preimage {Z : Set α} (hZ : IsPreirreducible Z) {f : β
 #align is_preirreducible.preimage IsPreirreducible.preimage
 
 end Preirreducible
+
+section codiscrete
+
+/-- The empty subset of any topological space is a discrete set in its
+subspace topology. -/
+lemma discreteTopology_emptyset : DiscreteTopology ↑(∅ : Set α) :=
+  discreteTopology_iff_nhds.mpr (fun x ↦ False.elim x.property)
+
+/-- We say a set `U` in a topological space `α` is *codiscrete* if it is
+open with discrete complement. -/
+def IsCodiscrete (U : Set α) : Prop := IsOpen U ∧ DiscreteTopology ↑Uᶜ
```

Comment:
Do you have any use case in a non-regular setting? If not, I think we could live with only the definition of the filter, even if some of the properties hold more generally. Except if you want a separate definition specifically for dot notation?

Reply:
The only use case I have is in the setting of open subsets of `ℂ`, I just thought it was mathlib convention to make every individual lemma as general as possible. 

I'm happy to inline the monotonicity, etc, into the definition of the filter, although I would prefer to keep "discreteTopology_union" as a separate lemma.